manticore/core/executor.py

```diff
@@ -487,9 +487,20 @@ def generate_testcase(self, state, message = 'Testcase generated'):
         assert solver.check(state.constraints)
         for symbol in state.input_symbols:
             buf = solver.get_value(state.constraints, symbol)
-            file(self._getFilename('test_%08x.txt'%test_number),'a').write("%s: %s\n"%(symbol.name, repr(buf)))
+            open(self._getFilename('test_{:08x}.txt'.format(test_number)),'a').write("{:s}: {:s}\n".format(symbol.name, repr(buf)))
         
-        file(self._getFilename('test_%08x.syscalls'%test_number),'a').write(repr(state.platform.syscall_trace))
+        open(self._getFilename('test_{:08x}.syscalls'.format(test_number)),'a').write(repr(state.platform.syscall_trace))
+
+        # save symbolic files
+        files = getattr(state.platform, 'files', None)
```

Comment:
all these getattr checks doesn't feel very pythonic to me. i think we might want to also move this symbolic file saving code into the platform 
1. to avoid copypasting this code to save symbolic files
2. because the executor/state shouldn't be digging too much into the internals of the platform? (.files)
3. because we could remove the getattrs since can be certain that .files and .array exists, etc

however, i think the platform shouldn't necessarily be the one to touch disk (or whatever storage backends we may implement in the future), maybe it should just provide the data and its name (i.e. filename) and return back to the executor or state for writing.

thoughts?